elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/TableExportIT.java

```diff
@@ -0,0 +1,1441 @@
+            } else if (!(responseGraphQL.contains("\"status\":\"PROCESSING\""))) {
+                fail("TableExport has failed.");
+                break;
+            }
+            i++;
+
+            if (i == 1000) {
+                fail("TableExport not completed.");
+            }
+        }
+    }
+
+    /**
+     * Test for a GraphQL query as a Table Export Request with Relationship fetch.
+     * @throws InterruptedException InterruptedException
+     */
+    @Test
+    public void graphQLRelationshipFetchFail() throws InterruptedException {
+
+        AsyncDelayStoreTransaction.sleep = true;
+        TableExport queryObj = new TableExport();
+        queryObj.setId("edc4a871-dbf2-4054-804e-d80075cf828e");
+        queryObj.setAsyncAfterSeconds(0);
+        queryObj.setQueryType("GRAPHQL_V1_0");
+        queryObj.setStatus("QUEUED");
+        queryObj.setResultType("CSV");
+        queryObj.setQuery("{\"query\":\"{ book { edges { node { title authors {edges { node { name } } } } } } }\", \"variables\":null}");
+        String graphQLRequest = document(
+                 mutation(
+                         selection(
+                                 field(
+                                         "tableExport",
+                                         arguments(
+                                                 argument("op", "UPSERT"),
+                                                 argument("data", queryObj, UNQUOTED_VALUE)
+                                         ),
+                                         selections(
+                                                 field("id"),
+                                                 field("query"),
+                                                 field("queryType"),
+                                                 field("status"),
+                                                 field("resultType")
+                                         )
+                                 )
+                         )
+                 )
+        ).toQuery();
+
+        JsonNode graphQLJsonNode = toJsonNode(graphQLRequest, null);
+        ValidatableResponse response = given()
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .body(graphQLJsonNode)
+                .post("/graphQL")
+                .then()
+                .statusCode(org.apache.http.HttpStatus.SC_OK);
+
+        String expectedResponse = "{\"data\":{\"tableExport\":{\"edges\":[{\"node\":{\"id\":\"edc4a871-dbf2-4054-804e-d80075cf828e\","
+                + "\"query\":\"{\\\"query\\\":\\\"{ book { edges { node { title authors {edges { node { name } } } } } } }\\\", \\\"variables\\\":null}\","
+                + "\"queryType\":\"GRAPHQL_V1_0\",\"status\":\"PROCESSING\",\"resultType\":\"CSV\"}}]}}}";
+        assertEquals(expectedResponse, response.extract().body().asString());
+
+        int i = 0;
+        while (i < 1000) {
```

Comment:
We need to stop repeating this test structure over and over - and generalize this somehow.